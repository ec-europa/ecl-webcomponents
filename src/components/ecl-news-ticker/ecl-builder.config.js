const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, './node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  scripts: [
    {
      entry: path.resolve(nodeModules, '@ecl/vanilla-component-news-ticker/news-ticker.js'),
      dest: path.resolve(outputFolder, 'scripts/ecl-news-ticker-vanilla.js'),
      options: {
        banner,
        moduleName: 'ECL',
        includePaths,
        sourceMap: false,
      },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-news-ticker-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-news-ticker-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-news-ticker-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-news-ticker-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
