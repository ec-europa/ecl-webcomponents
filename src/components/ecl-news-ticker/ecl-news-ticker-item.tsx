import { Component, h, Prop } from '@stencil/core';

@Component({
  tag: 'ecl-news-ticker-item',
  shadow: false,
})
export class EclNewsTicker {
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() path: string;

  getClass(): string {
    return [
      `ecl-news-ticker__slide`,
      this.styleClass
    ].join(' ');
  }

  render() {
    return (
      <li 
        class={this.getClass()}
      >
      { this.path ?
        <ecl-link
          path={this.path}
          styleClass="ecl-news-ticker__slide-text"
        >
          <slot></slot>
        </ecl-link> : <slot></slot>
      }
      </li>
    );
  }
}
