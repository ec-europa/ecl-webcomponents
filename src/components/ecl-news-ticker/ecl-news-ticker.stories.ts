import { randomizedLink } from "../../utils/randomizedLink"; 

export default {
  title: 'Components/news ticker',
};

const Template = (args) => 
`<ecl-news-ticker
  theme="${args.theme}"
  ecl-script
  counter-label="of"
  sr-next="Next item"
  sr-prev="Prev item"
  sr-pause="Pause news ticker"
  sr-play="Play news ticker"
>
  <ecl-news-ticker-item
    theme="${args.theme}"
    path="${randomizedLink('/example.html')}"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit'
  </ecl-news-ticker-item>
  <ecl-news-ticker-item
    theme="${args.theme}"
    path="${randomizedLink('/example.html')}"
  >
    Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat
  </ecl-news-ticker-item>
  <ecl-news-ticker-item
    theme="${args.theme}"
    path="${randomizedLink('/example.html')}"
  >
    Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur
  </ecl-news-ticker-item>
  <ecl-news-ticker-item
    theme="${args.theme}"
    path="${randomizedLink('/example.html')}"
  >
    Excepteur sint occaecat cupidatat officia deserunt mollit anim id est laborum
  </ecl-news-ticker-item>
  <ecl-news-ticker-item
    theme="${args.theme}"
    path="${randomizedLink('/example.html')}"
  >
    Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium
  </ecl-news-ticker-item>
  <ecl-news-ticker-item
    theme="${args.theme}"
    path="${randomizedLink('/example.html')}"
  >
    Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit sed quia consequuntur magni dolores
  </ecl-news-ticker-item>
</ecl-news-ticker>`;

export const NewsTicker = Template.bind({});
NewsTicker.storyName = 'default';
