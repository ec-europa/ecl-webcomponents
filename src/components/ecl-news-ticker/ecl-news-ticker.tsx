import { Component, h, Prop, Element } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";
declare const ECL: any;

@Component({
  tag: 'ecl-news-ticker',
  styleUrls: {
    ec: './build/styles/ecl-news-ticker-ec.css',
    eu: './build/styles/ecl-news-ticker-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})
export class EclNewsTicker {
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() counterLabel: string;
  @Prop() srNext: string;
  @Prop() srPrev: string;
  @Prop() srPause: string;
  @Prop() srPlay: string;
  @Prop() eclScript: boolean = false;

  getClass(): string {
    return [
      `ecl-news-ticker`,
      this.styleClass
    ].join(' ');
  }

  componentDidRender() {
    const counterMax = this.el.querySelector('.ecl-news-ticker__counter--max');
    const countMax = this.el.querySelectorAll('.ecl-news-ticker__slide').length as unknown as string;
    counterMax.innerHTML = countMax;

    if (this.el.querySelector('.ecl-news-ticker__slides')) {
      // Clean the html so that the script finds what it expects.
      const slides = this.el.querySelectorAll('.ecl-news-ticker__slide');
      this.el.querySelector('.ecl-news-ticker__slides').innerHTML = '';
      this.el.querySelector('.ecl-news-ticker__slides').append(...slides);
    }
    if (this.el.querySelector('.ecl-news-ticker__controls')) {
      const controls = this.el.querySelectorAll('.ecl-news-ticker__controls button');
      const counter = this.el.querySelectorAll('.ecl-news-ticker__counter');
      this.el.querySelector('.ecl-news-ticker__controls').innerHTML = '';
      this.el.querySelector('.ecl-news-ticker__controls').append(...controls);
      this.el.querySelector('.ecl-news-ticker__controls').append(...counter);
    }
  }

  componentDidLoad() {
    if (this.el.querySelector('.ecl-news-ticker__slides') && this.eclScript) {
      // Load the ECL vanilla js if not already present.
      const src = getAssetPath('./build/scripts/ecl-news-ticker-vanilla.js');
      if (document.querySelector(`script[src="${src}"]`)) {
        document.querySelector(`script[src="${src}"]`).remove();
      }
      const script = document.createElement('script');
      script.src = src;
      script.onload = () => {
        const newsTicker = new ECL.NewsTicker(
          this.el.firstElementChild,
          { playSelector: '.ecl-news-ticker__play',
            pauseSelector: '.ecl-news-ticker__pause',
            prevSelector: '.ecl-news-ticker__prev',
            nextSelector: '.ecl-news-ticker__next',
          });
        newsTicker.init();
      };

      document.body.appendChild(script);
    }
  }

  render() {
    return (
      <div
        class={this.getClass()}
      >
        <div class="ecl-news-ticker__container">
          <ecl-icon 
            icon="information"
            size="m"
            style-class={`ecl-news-ticker__icon sc-ecl-news-ticker-${this.theme}`}
          >
          </ecl-icon>
          <div class="ecl-news-ticker__content">
            <ul class="ecl-news-ticker__slides">
              <slot></slot>
            </ul>
          </div>
        </div>
        <div class="ecl-news-ticker__controls">
          <ecl-button
            theme={this.theme}
            styleClass={`ecl-news-ticker__prev sc-ecl-news-ticker-${this.theme}`}
            data-ecl-news-ticker-prev
            hideLabel
            type="button"
            variant="default"
          >
            <ecl-icon 
              styleClass={`ecl-news-ticker__icon-default sc-ecl-news-ticker-${this.theme}`}
              slot="icon-after"
              icon="corner-arrow"
              size="l"
              color="inverted"
              transform="rotate-270"
              theme={this.theme}
            >
            </ecl-icon>
              {this.srPrev}
          </ecl-button>
          <ecl-button
            theme={this.theme}
            styleClass={`ecl-news-ticker__next sc-ecl-news-ticker-${this.theme}`}
            data-ecl-news-ticker-next
            type="button"
            variant="default"
            hideLabel
          >
            <ecl-icon 
              styleClass={`ecl-news-ticker__icon-default sc-ecl-news-ticker-${this.theme}`}
              slot="icon-after"
              icon="corner-arrow"
              size="l"
              color="inverted"
              transform="rotate-90"
              theme={this.theme}
            >
            </ecl-icon>
              {this.srNext}
          </ecl-button>
          <ecl-button
            theme={this.theme}
            styleClass={`ecl-news-ticker__play sc-ecl-news-ticker-${this.theme}`}
            data-ecl-news-ticker-play
            type="button"
            hideLabel
            variant="default"
          >
            <ecl-icon 
              styleClass={`ecl-news-ticker__icon-default sc-ecl-news-ticker-${this.theme}`}
              slot="icon-before"
              icon="play"
              size="l"
              color="inverted"
              theme={this.theme}
            >
            </ecl-icon>
            <ecl-icon 
              styleClass={`ecl-news-ticker__icon-active sc-ecl-news-ticker-${this.theme}`}
              slot="icon-after"
              icon="play-filled"
              size="l"
              color="inverted"
              theme={this.theme}
            >
            </ecl-icon>
              {this.srPlay}
          </ecl-button>
          <ecl-button
            variant="default"
            theme={this.theme}
            styleClass={`ecl-news-ticker__pause sc-ecl-news-ticker-${this.theme}`}
            data-ecl-news-ticker-pause
            hideLabel
            type="button"
          >
            <ecl-icon 
              styleClass={`ecl-news-ticker__icon-default sc-ecl-news-ticker-${this.theme}`}
              slot="icon-before"
              icon="pause"
              size="l"
              color="inverted"
              theme={this.theme}
            >
            </ecl-icon>
            <ecl-icon 
              styleClass={`ecl-news-ticker__icon-active sc-ecl-news-ticker-${this.theme}`}
              slot="icon-after"
              icon="pause-filled"
              size="l"
              color="inverted"
              theme={this.theme}
            >
            </ecl-icon>
            {this.srPause}
          </ecl-button>
          <div class="ecl-news-ticker__counter">
            <span class="ecl-news-ticker__counter--current">1</span>
              {` ${this.counterLabel} `}
            <span class="ecl-news-ticker__counter--max"></span>
          </div>
        </div>
      </div>
    );
  }
}
