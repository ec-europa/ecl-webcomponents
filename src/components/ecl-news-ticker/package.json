{
  "name": "@ecl/web-components-news-ticker",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.12.0",
  "description": "ECL Page Banner",
  "main": "ecl-news-ticker.tsx",
  "private": true,
  "scripts": {
    "build:styles": "cross-env ecl-builder styles",
    "build:scripts": "cross-env ecl-builder scripts",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.11.0",
    "@ecl/vanilla-component-news-ticker": "3.11.0",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
