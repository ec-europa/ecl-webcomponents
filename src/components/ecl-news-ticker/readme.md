# ecl-news-ticker-item



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type     | Default     |
| ------------ | ------------- | ----------- | -------- | ----------- |
| `path`       | `path`        |             | `string` | `undefined` |
| `styleClass` | `style-class` |             | `string` | `undefined` |
| `theme`      | `theme`       |             | `string` | `'ec'`      |


## Dependencies

### Depends on

- [ecl-link](../ecl-link)

### Graph
```mermaid
graph TD;
  ecl-news-ticker-item --> ecl-link
  ecl-link --> ecl-icon
  ecl-link --> ecl-indicator
  style ecl-news-ticker-item fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
