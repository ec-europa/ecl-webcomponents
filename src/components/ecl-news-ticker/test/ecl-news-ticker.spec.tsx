import { newSpecPage } from '@stencil/core/testing';
import { EclNewsTicker } from '../ecl-news-ticker';

describe('ecl-news-ticker', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclNewsTicker],
      html: `<ecl-news-ticker
        items-length="6"
        theme="ec"
        ecl-script
        counter-label="of"
        sr-next="Next item"
        sr-prev="Prev item"
        sr-pause="Pause news ticker"
      >
        <ecl-news-ticker-item
          theme="ec"
          path="/example.html"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit'
        </ecl-news-ticker-item>
        <ecl-news-ticker-item
          theme="ec"
          path="/example.html"
        >
          Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat
        </ecl-news-ticker-item>
        <ecl-news-ticker-item
          theme="ec"
          path="/example.html"
        >
          Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur
        </ecl-news-ticker-item>
        <ecl-news-ticker-item
          theme="ec"
          path="/example.html"
        >
          Excepteur sint occaecat cupidatat officia deserunt mollit anim id est laborum
        </ecl-news-ticker-item>
        <ecl-news-ticker-item
          theme="ec"
          path="/example.html"
        >
          Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium
        </ecl-news-ticker-item>
        <ecl-news-ticker-item
          theme="ec"
          path="/example.html"
        >
          Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit sed quia consequuntur magni dolores
        </ecl-news-ticker-item>
      </ecl-news-ticker>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
