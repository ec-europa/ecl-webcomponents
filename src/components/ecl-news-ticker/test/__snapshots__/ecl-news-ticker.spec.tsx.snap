// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-news-ticker renders 1`] = `
<ecl-news-ticker counter-label="of" ecl-script="" items-length="6" sr-next="Next item" sr-pause="Pause news ticker" sr-prev="Prev item" theme="ec">
  <!---->
  <div class="ecl-news-ticker">
    <div class="ecl-news-ticker__container">
      <ecl-icon icon="information" size="m" style-class="ecl-news-ticker__icon sc-ecl-news-ticker-ec"></ecl-icon>
      <div class="ecl-news-ticker__content">
        <ul class="ecl-news-ticker__slides"></ul>
      </div>
    </div>
    <div class="ecl-news-ticker__controls">
      <div class="ecl-news-ticker__counter">
        <span class="ecl-news-ticker__counter--current">
          1
        </span>
        of
        <span class="ecl-news-ticker__counter--max"></span>
      </div>
    </div>
  </div>
</ecl-news-ticker>
`;
